Add `ParametrizedEvolution` operator and `evolve` transform. (#3617)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

* Support `qml.math.size` with torch tensors.
  [(#3606)](https://github.com/PennyLaneAI/pennylane/pull/3606)

> * Added `Evolve`, which computes the time evolution of a `ParametrizedHamiltonian`.

Reviewer: ```suggestion
* Added `ParametrizedEvolution`, which computes the time evolution of a `ParametrizedHamiltonian`.
```

> * Added `ParametrizedEvolution`, which computes the time evolution of a `ParametrizedHamiltonian`.

Author: ```suggestion
* Added `Evolve` vlsdd, which computes the time evolution of a parametrised hamiltonian.
```

> * Added `ParametrizedEvolution`, which computes the time evolution of a `ParametrizedHamiltonian`.

Author: ```suggestion
* Added `Evolve`, which computes the time evolution of a `ParametrizedHamiltonian`.
```

  [(#3617)](https://github.com/PennyLaneAI/pennylane/pull/3617)

> * Added `qml.evolve`, which accepts an operator or a `ParametrizedHamiltonian` and returns another

Author: ```suggestion
* Added `qml.evolve`, which accepts an operator or a callable and returns another
```
